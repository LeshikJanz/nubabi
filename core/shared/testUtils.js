import configureMockStore from 'redux-mock-store';
import { combineEpics, createEpicMiddleware } from 'redux-observable';

export const setupEpic = (epic) => {
  const epicMiddleware = createEpicMiddleware(epic);
  const mockStore = configureMockStore([epicMiddleware]);

  return {
    middleware: epicMiddleware,
    mockStore: mockStore,
    afterEach: () => {
      epicMiddleware.replaceEpic(epic);
    }
  }
};
