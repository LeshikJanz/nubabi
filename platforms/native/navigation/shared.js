import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) =>
  pathOr('Nubabi', ['params', 'babyName'], state);

export function getBabyTitle(navigation) {
  if (!navigation) {
    throw new Error('an screen hasnt been updated, check stack');
  }

  return babyNameTitle(navigation);
}

export const getTabHeaders = navigate => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
