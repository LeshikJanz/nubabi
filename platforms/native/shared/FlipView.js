// @flow
import React, { Component } from 'react';
import { View, Animated } from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

type Props = {
  renderFront: Function,
  renderBack: Function,
  velocity: number,
  tension: number,
  friction: number,
};

export class FlipCard extends Component {
  props: Props;

  state = {
    animatedValue: new Animated.Value(0),
    isFlipped: true,
  };

  componentDidUpdate(prevProps: Props, prevState: *) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard = () => {
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0, // Returns to the start
      velocity: this.props.velocity, // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction, // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront(this._flipToggleCard);
    }
    return this.props.renderBack(this._flipToggleCard);
  }

  render() {
    const perspective = 2000;
    const rotateY = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <View style={styles.animatedContainer}>
        <Animated.View
          style={[
            styles.animatedContainer,
            { transform: [{ rotateY }, { perspective }] },
          ]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </View>
    );
  }
}

export default FlipCard;
