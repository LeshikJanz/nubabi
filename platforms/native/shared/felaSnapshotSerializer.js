// WIP
// prettier-ignore
const felaSnapshotSerializer = {
  test(val) {
    return val && val.felaRules && val.$$typeof === Symbol.for('react.test.json');
  },

  print(val, print) {
    const styles = val.felaRules;
    val.felaRules = false;

    return `style = Object ${JSON.stringify(styles, null, 4)}\n\n${print(val)}`;
  },
};

module.exports = felaSnapshotSerializer;
