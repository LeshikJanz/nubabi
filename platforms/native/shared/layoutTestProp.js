export default {
  fontScale: 2,
  height: 1334,
  scale: 2,
  width: 750,
  viewportWidth: 750,
  viewportHeight: 1334,
};
