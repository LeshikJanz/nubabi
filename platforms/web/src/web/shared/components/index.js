import About from "./About";
import AsyncComponent from "./AsyncComponent";
import AuthenticatedRoute from "./AuthenticatedRoute";
import Header from "./Header";
import Home from "./Home";
import Loader from "./Loader";
import Test from "./Test";

export default {
  About,
  AsyncComponent,
  AuthenticatedRoute,
  Header,
  Home,
  Loader,
  Test
};
