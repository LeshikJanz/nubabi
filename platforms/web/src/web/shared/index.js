import * as actions from "./actions";
export { default as containers } from "./containers";
export { default as reducer } from "./reducer";
export { default as epics } from "./epics";
export { actions };
