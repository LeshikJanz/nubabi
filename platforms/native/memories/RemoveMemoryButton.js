// @flow
import type { State } from 'core/types';
import React, { PureComponent } from 'react';
import { Alert, TouchableOpacity } from 'react-native';
import { compose } from 'ramda';
import { gql, graphql } from 'react-apollo';
import { connect } from 'react-redux';
<<<<<<< HEAD:src/native/memories/RemoveMemoryButton.js
import { removeEdgeFromFragment } from '../../common/helpers/graphqlUtils';
import { Text } from '../components';
import RecentMemories from '../profile/RecentMemories';
import { ViewMemories } from './ViewMemories';
import { appError } from '../../common/app/actions';
import { toggleNetworkActivityIndicator } from '../../common/ui/reducer';
=======
import { removeEdgeFromFragment } from 'core/helpers/graphqlUtils';
import { Icon } from '../components';
import RecentMemories from '../profile/RecentMemories';
import { ViewMemories } from './ViewMemories';
import { appError } from 'core/app/actions';
import { toggleNetworkActivityIndicator } from 'core/ui/reducer';
>>>>>>> refactor: module aliases:platforms/native/memories/EditMemoryHeader.js

type Props = {
  goBack: () => void,
  appError: (error: Error) => void,
  onSubmit: () => Promise<ApolloQueryResult<*>>,
};

class RemoveMemoryButton extends PureComponent {
  props: Props;

  promptForDeletion = () => {
    Alert.alert(
      'Delete Memory',
      'Are you sure you want to delete this memory?',
      [
        {
          text: 'Cancel',
          style: 'cancel',
        },
        {
          text: 'Yes',
          onPress: this.handleDelete,
        },
      ],
    );
  };

  handleDelete = () => {
    this.props.onSubmit().catch(() =>
      setTimeout(() => {
        this.props.appError(
          new Error(
            'There was a problem deleting a memory. Please try again later.',
          ),
        );
      }, 1000),
    );
    this.props.goBack();
  };

  render() {
    return (
      <TouchableOpacity
        onPress={this.promptForDeletion}
        style={{
          flex: 1,
          marginRight: 10,
          padding: 16,
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Text color="primary" bold>
          delete this memory
        </Text>
      </TouchableOpacity>
    );
  }
}

export default compose(
  connect(
    ({ babies }: State) => ({
      currentBabyId: babies.currentBabyId,
    }),
    { appError, toggleNetworkActivityIndicator },
  ),
  graphql(
    gql`
      mutation DeleteMemory($input: DeleteMemoryInput!) {
        deleteMemory(input: $input) {
          memory {
            id
          }
        }
      }
    `,
    {
      props: ({ mutate, ownProps }) => ({
        onSubmit: () => {
          ownProps.toggleNetworkActivityIndicator(true);

          // $FlowFixMe$
          return mutate({
            variables: { input: { id: ownProps.id } },
            optimisticResponse: {
              __typename: 'Mutation',
              deleteMemory: {
                __typename: 'DeleteMemoryPayload',
                memory: {
                  __typename: 'Memory',
                  id: ownProps.id,
                },
              },
            },
            update: (store, data) => {
              const options = [
                ownProps.id,
                ownProps.currentBabyId,
                ['memories'],
              ];

              removeEdgeFromFragment(
                RecentMemories.fragments.memories,
                ...options,
              )(store, data);

              removeEdgeFromFragment(ViewMemories.fragments.list, ...options, {
                fragmentName: 'Memories',
              })(store, data);
            },
          }).finally(() => ownProps.toggleNetworkActivityIndicator(false));
        },
      }),
    },
  ),
)(RemoveMemoryButton);
