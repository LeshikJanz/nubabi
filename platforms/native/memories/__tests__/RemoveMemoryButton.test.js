import 'react-native';
import React from 'react';
import { RemoveMemoryButton } from '../RemoveMemoryButton';
import { expectRender } from '../../shared/testUtils';

describe('RemoveMemoryButton', () => {
  it('renders correctly', () => {
    expectRender(
      <RemoveMemoryButton

      />,
    );
  });
});
