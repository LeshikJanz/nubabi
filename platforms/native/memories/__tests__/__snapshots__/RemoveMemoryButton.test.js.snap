// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveMemoryButton renders correctly 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#EA3154",
    "fontFamily": "SF Pro Text",
    "fontWeight": "700"
}

<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginRight": 10,
      "opacity": 1,
      "padding": 16,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "color": "#EA3154",
        "fontFamily": "SF Pro Text",
        "fontSize": 13,
        "fontWeight": "700",
        "lineHeight": 16,
      }
    }
  >
    delete this memory
  </Text>
</View>
`;
